plugins/ua_config_file_based.c

```diff
@@ -0,0 +1,575 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information.
+ *
+ *    Copyright 2023 (c) Fraunhofer IOSB (Author: Noel Graf)
+ */
+
+#include <open62541/server_config_file_based.h>
+#include <open62541/plugin/log_stdout.h>
+#include "cj5.h"
+
+#define MAX_TOKENS 256
+
+typedef struct {
+    const char *json;
+    const cj5_token *tokens;
+    cj5_result result;
+    unsigned int tokensSize;
+    size_t index;
+    UA_Byte depth;
+} ParseCtx;
+
+static UA_ByteString
+getJsonPart(cj5_token tok, const char *json) {
+    unsigned int str_len = (tok.end - tok.start) + 1;
+    if(tok.type == CJ5_TOKEN_STRING) {
```

Comment:
Don’t allocate new memory.
You can just return a pointer into the json string.

You don‘t need to add the zero termination.
The token contains the length. You can use that directly for the returned bytestring.